modules/openapi-generator/src/main/resources/go/api.mustache

```diff
@@ -290,7 +290,7 @@ func (a *{{{classname}}}Service) {{{nickname}}}(ctx context.Context{{#hasParams}
 		}
 		{{#responses}}
 		{{#dataType}}
-		if localVarHttpResponse.StatusCode == {{{code}}} {
+		{{^wildcard}}if localVarHttpResponse.StatusCode == {{{code}}} { {{/wildcard}}
```

Comment:
This counts on the fact that the wildcard response will be the last one. My (limited) testing confirmed this, but will that always be the case?

Reply:
Ideally this should catch any error code so the error model is returned if one is specified. It may need to catch the default/wildcard in these cases.

If i remember correct, this should be an interface so it could be possible to make this completely generic without specifically checking each code?

Reply:
> Ideally this should catch any error code so the error model is returned if one is specified. It may need to catch the default/wildcard in these cases.

My approach here was supressing the status code check for the default case so that it handles any error code. But that may overcatch if the wildcard response isn't the last one. Hence my original question.

> If i remember correct, this should be an interface so it could be possible to make this completely generic without specifically checking each code?

Sorry, I don't understand. Are you suggesting removing that check completely and just trying to unmarshall the error model based on `datatype`?